### PR TITLE
Make tgmath header C compliant

### PR DESCRIPTION
**Pull request by memontic-ms**
Previously, .m files that imported tgmath.h could not be compiled as Objective C. Since tgmath.h included the C++ header cmath, .m files had to be compiled as Objective C++.

Fix: #735

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/1466)

### CODE REVIEW

File: msvc/LLVM-3.6.0/lib/clang/3.7.0/include/tgmath.h [hunk 1]
```diff
@@ -29,7 +29,11 @@
 #define __TGMATH_H
 
 /* C99 7.22 Type-generic math <tgmath.h>. */
+#if defined(__cplusplus)
```
Comment (aballway) [inline msvc/LLVM-3.6.0/lib/clang/3.7.0/include/tgmath.h]: should the copyright banner be updated?

Comment (memontic-ms) [inline msvc/LLVM-3.6.0/lib/clang/3.7.0/include/tgmath.h]: It's not a Microsoft copyright banner. What would need updating?

**Comment by DHowett-MSFT** [comment]
I'm concerned that when we get another compiler header/lib drop we'll stomp this change sight unseen.

**Comment by memontic-ms** [comment]
@DHowett-MSFT I have included ObjCImportTest.m within the Clang unit tests that will catch a regression on this issue for future compiler drops.
Related, I have renamed the ClangModules unit tests project to just Clang since the added test file is not dependent on modules support.

**Pull request merged**